src/parser.js

```diff
@@ -365,13 +377,13 @@ export default class Parser {
         return parameter.in === 'query';
       });
 
-      const basePath = (this.swagger.basePath || '').replace(/[/]+$/, '');
-      const hrefForResource = buildUriTemplate(basePath, href, resourceParameters, queryParameters);
-      // TODO: This is probably a bug. Transitions can have their own `href` and
-      // we should use that instead of resetting the resource's href each time.
-      // The open question is do resources still get an href? Then we should
-      // only set it here on the transition if it is different.
-      resource.attributes.set('href', hrefForResource);
+      // Here we generate a URI template specific to this transition. If it
+      // is different from the resource URI template, then we set the
+      // transition's `href` attribute.
+      const hrefForTransition = buildUriTemplate(this.basePath, href, resourceParameters, queryParameters);
```

Comment:
Does this add any parameters from the resource to this? In other words, a resource can have X query parameters and a transition can have X + Y query parameters.